plugins/heroku/src/herokuClient.ts

```diff
@@ -1,5 +1,27 @@
-import Heroku from 'heroku-client'
+import { ToolKitError } from '@dotcom-tool-kit/error'
+import { styles } from '@dotcom-tool-kit/logger'
+import Heroku, { HerokuError } from 'heroku-client'
 
 const HEROKU_AUTH_TOKEN = process.env.HEROKU_AUTH_TOKEN
 
 export default new Heroku({ token: HEROKU_AUTH_TOKEN })
+
+// the Heroku client doesn't give particularly useful error messages, so we require an additional context
+// string when handling errors coming from it
+export function extractHerokuError(context: string): (err: unknown) => never {
+  return (err) => {
+    if ((err as HerokuError).statusCode) {
```

Comment:
**suggestion**: this would be safer as an `instanceof HerokuError` check (which would also save you from having to do `const upstream = err as HerokuError`)